test/decoder_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule MozartFetcher.DecoderTest do
+  use ExUnit.Case
+
+  alias MozartFetcher.{Decoder, Config, Envelope}
+
+  doctest Decoder
+
+  describe "#data_to_struct" do
+    test "when the JSON is successfully decoded it is transformed into a struct" do
+      json = ~s({"head":[],"bodyInline":"<DIV id=\\"site-container\\">","bodyLast":[]})
+
+      expected = {:ok, %Envelope{
+        head: [],
+        bodyInline: ~s(<DIV id="site-container">),
+        bodyLast: []
+      }}
+
+      assert Decoder.data_to_struct(json, %Envelope{}) == expected
+    end
+  end
+
+  describe "#list_to_struct_list" do
+    test "when the list of JSON is successfuly decoded it is transformed into a list of structs" do
+      json =  ~s({ "components\": [{"id": "stream-icons", "endpoint": "localhost:8082/success", "must_succeed": true },
+                                  {"id": "stream-icons", "endpoint": "localhost:8082/success", "must_succeed": true }
```

Comment:
Can we set the second one to have different values in case we are transforming the same data twice? It looks to iterate on each one, but just in case.